crucible/src/Lang/Crucible/Backend/ProofGoals.hs

```diff
@@ -210,6 +210,23 @@ gcAddGoals g (TopCollector gs) = TopCollector (gs Seq.|> g)
 gcAddGoals g (CollectingGoals gs gc) = CollectingGoals (gs Seq.|> g) gc
 gcAddGoals g gc = CollectingGoals (Seq.singleton g) gc
 
+-- | Add an assumption that is in scope for all goals.
```

Comment:
Perhaps say something like "... in scope for all goals, even ones that were added in earlier frames."